### ISSUE TITLE
Typescript: Don't flag type-only imports as duplicate imports

### ISSUE THREAD

**Issue by SebDuf** [rslint/rslint#124]
**I tried this code**: TypeScript supports import type-only information with a different syntax to make it easier to remove imports that produce no code during compilation. All of our codebases using this syntax, but it currently pollutes the output of rslint

A.ts
```ts
export type TypeA = string;

export const a: TypeA = 'something';
```

Test.ts
```ts
import type { TypeA } from './A';
import { a } from './A';

const b: TypeA = 'test';

console.log(a, b);
```

**I expected this to happen**: File passes lint successfully

**Instead this happened:**:  Imports were flagged as duplicate:

```
error[no-duplicate-imports]: `./A` import is duplicated
  ┌─ src/test.ts:2:19
  │
1 │ import type { TypeA } from './A';
  │                            ----- `./A` is first used here
2 │ import { a } from './A';
  │                   ^^^^^ `./A` is then used again here

Outcome: 1 fail, 0 warn, 0 success
```

### Meta 
`rslint -V`: 0.3.1

**Additional context**
Type-only imports: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#type-only-imports-and-export
